src/metabase/email/messages.clj

```diff
@@ -226,3 +226,121 @@
                        (vec (cons :div (for [result results]
                                          (render/render-pulse-section timezone result)))))]
     (render-message-body "metabase/email/pulse" (pulse-context body pulse) (seq @images))))
+
+(defn pulse->alert-condition-kwd
+  "Given an `ALERT` return a keyword representing what kind of goal needs to be met."
+  [{:keys [alert_above_goal alert_condition card creator] :as alert}]
+  (if (= "goal" alert_condition)
+    (if (true? alert_above_goal)
+      :meets
+      :below)
+    :rows))
+
+(defn- first-card
+  "Alerts only have a single card, so the alerts API accepts a `:card` key, while pulses have `:cards`. Depending on
+  whether the data comes from the alert API or pulse tasks, the card could be under `:card` or `:cards`"
+  [alert]
+  (or (:card alert)
+      (first (:cards alert))))
+
+(defn- default-alert-context
+  ([alert]
+   (default-alert-context alert nil))
+  ([alert alert-condition-map]
+   (let [{card-id :id, card-name :name} (first-card alert)]
+     (merge {:questionURL (url/card-url card-id)
+             :questionName card-name
+             :emailType    "alert"
+             :sectionStyle render/section-style
+             :colorGrey4   render/color-gray-4
+             :logoFooter   true}
+            (random-quote-context)
+            (when alert-condition-map
+              {:alertCondition (get alert-condition-map (pulse->alert-condition-kwd alert))})))))
+
+(defn- alert-results-condition-text [goal-value]
+  {:meets (format "reached its goal of %s" goal-value)
+   :below (format "gone below its goal of %s" goal-value)
+   :rows  "results for you to see"})
+
+(defn render-alert-email
+  "Take a pulse object and list of results, returns an array of attachment objects for an email"
+  [timezone {:keys [alert_first_only] :as alert} results goal-value]
+  (let [images       (atom {})
+        body         (binding [render/*include-title* true
+                               render/*render-img-fn* (partial render-image images)]
+                       (html (vec (cons :div (for [result results]
+                                               (render/render-pulse-section timezone result))))))
+        message-ctx  (default-alert-context alert (alert-results-condition-text goal-value))]
+    (render-message-body "metabase/email/alert"
+                         (assoc message-ctx :pulse body :firstRunOnly? alert_first_only)
+                         (seq @images))))
```

Comment:
it looks like about half of this function (the `images` and `body` part) is just copy-pasta from `render-pulse-email`. Maybe there's some way we could avoid repeating the logic between these two `render-` functions and have them both call a shared function that takes care of this stuff?